pkg/operation/botanist/component/nodeproblemdetector/node_problem_detector.go

```diff
@@ -0,0 +1,450 @@
+				Name:      serviceAccount.Name,
+				Namespace: serviceAccount.Namespace,
+			}},
+		}
+
+		clusterRolePSP = &rbacv1.ClusterRole{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:   clusterRolePSPName,
+				Labels: getLabels(),
+			},
+			Rules: []rbacv1.PolicyRule{
+				{
+					APIGroups:     []string{"extensions", "policy"},
+					Resources:     []string{"podsecuritypolicies"},
+					Verbs:         []string{"use"},
+					ResourceNames: []string{"node-problem-detector"},
+				},
+			},
+		}
+
+		clusterRoleBindingPSP = &rbacv1.ClusterRoleBinding{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:        clusterRoleBindingPSPName,
+				Annotations: map[string]string{resourcesv1alpha1.DeleteOnInvalidUpdate: "true"},
+				Labels:      getLabels(),
+			},
+			RoleRef: rbacv1.RoleRef{
+				APIGroup: rbacv1.GroupName,
+				Kind:     "ClusterRole",
+				Name:     clusterRolePSP.Name,
+			},
+			Subjects: []rbacv1.Subject{{
+				Kind:      rbacv1.ServiceAccountKind,
+				Name:      serviceAccount.Name,
+				Namespace: serviceAccount.Namespace,
+			}},
+		}
+
+		podSecurityPolicy = &policyv1beta1.PodSecurityPolicy{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:   podSecurityPolicyName,
+				Labels: getLabels(),
+			},
+			Spec: policyv1beta1.PodSecurityPolicySpec{
+				Privileged:               true,
+				AllowPrivilegeEscalation: pointer.Bool(true),
+				AllowedCapabilities: []corev1.Capability{
+					corev1.Capability(policyv1beta1.All),
+				},
+				Volumes: []policyv1beta1.FSType{
+					policyv1beta1.ConfigMap,
+					policyv1beta1.EmptyDir,
+					policyv1beta1.Projected,
+					policyv1beta1.Secret,
+					policyv1beta1.DownwardAPI,
+					policyv1beta1.HostPath,
+				},
+				HostNetwork: false,
+				HostIPC:     false,
+				HostPID:     false,
+				AllowedHostPaths: []policyv1beta1.AllowedHostPath{
+					{PathPrefix: "/etc/localtime"},
+					{PathPrefix: "/var/log"},
+					{PathPrefix: "/dev/kmsg"},
+				},
+				RunAsUser: policyv1beta1.RunAsUserStrategyOptions{
+					Rule: policyv1beta1.RunAsUserStrategyRunAsAny,
+				},
+				SELinux: policyv1beta1.SELinuxStrategyOptions{
+					Rule: policyv1beta1.SELinuxStrategyRunAsAny,
+				},
+				SupplementalGroups: policyv1beta1.SupplementalGroupsStrategyOptions{
+					Rule: policyv1beta1.SupplementalGroupsStrategyRunAsAny,
+				},
+				FSGroup: policyv1beta1.FSGroupStrategyOptions{
+					Rule: policyv1beta1.FSGroupStrategyRunAsAny,
+				},
+			},
+		}
+
+		daemonSet = &appsv1.DaemonSet{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      daemonSetName,
+				Namespace: metav1.NamespaceSystem,
+				Labels: map[string]string{
+					"app.kubernetes.io/instance": "shoot-core",
+					"app.kubernetes.io/name":     "node-problem-detector",
+					"origin":                     "gardener",
+					"gardener.cloud/role":        "system-component",
+				},
+			},
+			Spec: appsv1.DaemonSetSpec{
+				Selector: &metav1.LabelSelector{
+					MatchLabels: map[string]string{
+						"app":                        "node-problem-detector",
```

Comment:
```suggestion
						v1beta1constants.LabelApp:                        "node-problem-detector",
```

maybe add a `labelValueApp` for node-problem-detector too.